.github/workflows/tests.yaml

```diff
@@ -44,6 +44,30 @@ jobs:
           name: autograph-images-${{ github.sha }}
           path: docker-cache/
 
+  lint:
+    name: Linting
+    runs-on: ubuntu-22.04
+    permissions:
+      # Required: allow read access to the content for analysis.
+      contents: read
+      # Optional: allow read access to pull request. Use with `only-new-issues` option.
+      pull-requests: read
+      # Optional: allow write access to checks to allow the action to annotate code in the PR.
+      checks: write
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version: stable
```

Comment:
Rather than `stable` should we instead use the same version specified in the `go.mod`?

Reply:
I was hoping to not have another spot where we're updating the Go version across our codebase. The Go that runs the lint tooling being different than what we build the code seemed fine, given the [Go backwards compatibility](https://go.dev/blog/compat) guarantees.

If you're interested in this, I'm down, but it seemed like future noise and something we could lock down later if we truly needed to.

Reply:
Like, the violation pops up is actually because of the `Sprintf` call, but it doesn't go away when I do this work conditionally (because Sprintf doesn't actually care much). They're just pattern matching on "string + args". See this patch:

```
diff --git a/errors.go b/errors.go
index 9818b5b1..ba636d0b 100644
--- a/errors.go
+++ b/errors.go
@@ -23,7 +23,13 @@ func httpError(w http.ResponseWriter, r *http.Request, errorCode int, errorMessa
 		"code": errorCode,
 		"rid":  rid,
 	}).Errorf(errorMessage, args...)
-	msg := fmt.Sprintf(errorMessage, args...)
+	var msg string
+	if len(args) == 0 {
+		msg = errorMessage
+	} else {
+		msg = fmt.Sprintf(errorMessage, args...)
+	}
+
 	msg += "\r\nrequest-id: " + rid
 	// when nginx is in front of go, nginx requires that the entire
 	// request body is read before writing a response.
diff --git a/monitor_handler.go b/monitor_handler.go
index da97be8c..745ce17f 100644
--- a/monitor_handler.go
+++ b/monitor_handler.go
@@ -34,8 +34,7 @@ func (m *monitor) handleMonitor(w http.ResponseWriter, r *http.Request) {
 
 	for _, errstr := range m.sigerrstrs {
 		if errstr != "" {
-			// "%s" is a go vet false positive correction
-			httpError(w, r, http.StatusInternalServerError, "%s", errstr)
+			httpError(w, r, http.StatusInternalServerError, errstr)
 			return
 		}
 	}
```

Reply:
I think the core of my concern is the hypothetical scenario:
1. We develop our code against Go version X and the linter is very happy with our code.
2. Go releases a new version, X+1 as the new stable, and introduces a whole bunch of lints.
3. The golangci linter uses the stable version rather than our `go.mod` when deciding which lints apply.
4. Our CI now goes red even though we changed nothing in our code.

If the linter workflow examines our `go.mod` to determine the applicable lints, then I am fine leaving this PR as-is.

Reply:
Understood. I believe go.mod is looked at for some of the lints in `go vet` (e.g. the loop varialbe related ones). I'm less worried about breaking the CI cuz it's easy enough to plot down a specific Go version if it turns out we need it and like picking up new, better lints without having to work at it.

But I'm happy to adjust to my reviewer here.

Reply:
oh jeez, I had forgotten `go-version-file: 'path/to/go.mod'` existed. Just using that is clearly better. This is perhaps what you meant?